src/analyses/useAfterFree.ml

```diff
@@ -179,17 +175,15 @@ struct
     warn_exp_might_contain_freed "branch" ctx exp;
     ctx.local
 
-  let body ctx (f:fundec) : D.t =
-    ctx.local
-
   let return ctx (exp:exp option) (f:fundec) : D.t =
     Option.iter (fun x -> warn_exp_might_contain_freed "return" ctx x) exp;
     ctx.local
 
   let enter ctx (lval:lval option) (f:fundec) (args:exp list) : (D.t * D.t) list =
     let caller_state = ctx.local in
     List.iter (fun arg -> warn_exp_might_contain_freed "enter" ctx arg) args;
-    if AllocaVars.is_empty (fst caller_state) && HeapVars.is_empty (snd caller_state) then
+    [caller_state, (AllocaVars.empty (), snd caller_state)]
+  (* if AllocaVars.is_empty (fst caller_state) && HeapVars.is_empty (snd caller_state) then
```

Comment:
What's this commented-out code now?

Reply:
This is due to something that is left for future work in the UAF analysis:
* We used to consider only the caller state, reachable from the callee parameters
* However, we also need to consider everything from the caller state, reachable from the global program variables
* Problem is that we need a way to get a hold of the global vars
* Until then, we decided to simply keep the entire caller state as the callee state

I'll remove the commented-out portion and I'll add a TODO for the future work